azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -462,15 +465,27 @@ public void prefetchToken(final File tokenFile, final Props props, final Logger
     doPrefetch(tokenFile, props, logger, userToProxy);
   }
 
+  /*
+   * Create a suffix for Kerberos principal, the format is,
+   * az_<host name>_<execution id><DOMAIN_NAME>
+   */
+  private String kerberosSuffix(final Props props) {
+    final String host = ServiceProvider.SERVICE_PROVIDER.getInstance(Props.class)
+        .getString(AZKABAN_SERVER_HOST_NAME, "unknown");
+    final StringBuilder builder = new StringBuilder("az_");
+    builder.append(host);
+    builder.append("_");
+    builder.append(props.getString(FlowProperties.AZKABAN_FLOW_EXEC_ID));
+    builder.append(props.getString(HadoopSecurityManager.DOMAIN_NAME));
+    return builder.toString();
+  }
   private void doPrefetch(final File tokenFile, final Props props, final Logger logger,
       final String userToProxy) throws HadoopSecurityManagerException {
-    // Create suffix to be added to headless user, the suffix is a valid email ID which conforms
-    // to Kubernetes.
+    // Create suffix to be added to kerberos principal
     final String suffix =
-        props.getBoolean(HadoopSecurityManager.APPEND_SUBMIT_USER, false) &&
-            props.getString(HadoopSecurityManager.SUBMIT_USER_SUFFIX) != null ?
-            "/" + props.getString(Constants.FlowProperties.AZKABAN_FLOW_SUBMIT_USER) +
-                props.getString(HadoopSecurityManager.SUBMIT_USER_SUFFIX) : "";
```

Comment:
Why do we need to change the suffix?

Reply:
The requirement for suffix changed. The comment describes the new format.